File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/presentation_data_provider.java

package com.gracenote.gnsdk;

/** 
** Internal class  presentation_data_provider 
*/ 
 
public class presentation_data_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected presentation_data_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(presentation_data_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_presentation_data_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public presentation_data_provider() {
    this(gnsdk_javaJNI.new_presentation_data_provider__SWIG_0(), true);
  }

  public presentation_data_provider(GnMoodgridPresentationType type) {
    this(gnsdk_javaJNI.new_presentation_data_provider__SWIG_1(type.swigValue()), true);
  }

  public GnMoodgridDataPoint getData(long pos) {
    return new GnMoodgridDataPoint(gnsdk_javaJNI.presentation_data_provider_getData(swigCPtr, this, pos), true);
  }

  public long count() {
    return gnsdk_javaJNI.presentation_data_provider_count(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/presentation_data_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
** Internal class  presentation_data_provider
*/
public class presentation_data_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal presentation_data_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(presentation_data_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~presentation_data_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_presentation_data_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public presentation_data_provider() : this(gnsdk_csharp_marshalPINVOKE.new_presentation_data_provider__SWIG_0(), true) {
  }

  public presentation_data_provider(GnMoodgridPresentationType type) : this(gnsdk_csharp_marshalPINVOKE.new_presentation_data_provider__SWIG_1((int)type), true) {
  }

  public GnMoodgridDataPoint get_data(uint pos) {
    GnMoodgridDataPoint ret = new GnMoodgridDataPoint(gnsdk_csharp_marshalPINVOKE.presentation_data_provider_get_data(swigCPtr, pos), true);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.presentation_data_provider_count(swigCPtr);
    return ret;
  }

}

}
